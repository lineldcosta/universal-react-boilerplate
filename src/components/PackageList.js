/* @flow */
import React from 'react'

import type { Package } from '../actions/types'

type Props = {
  packages: Array<Package>,
  onRefresh: () => void
}

const PackageList = ({ packages, onRefresh }: Props) => (
  <div className='packages'>
    <button onClick={onRefresh.bind(this)}>Refresh</button>
    <ul>
      {packages.map((p: Package) => (
        <li>{ p.name }</li>
      ))}
    </ul>
  </div>
)

PackageList.propTypes = {
  packages: React.PropTypes.array.isRequired,
  onRefresh: React.PropTypes.func.isRequired
}

export default PackageList
