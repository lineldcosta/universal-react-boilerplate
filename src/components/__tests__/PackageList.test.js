/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import PackageList from '../PackageList'

const setup = (propOverrides) => {
  const props = Object.assign({
    onRefresh: () => null,
    packages: [
      {
        name: 'react',
        version: '15.4.2',
        description: 'React is a JavaScript library for building user interfaces.'
      },
      {
        name: 'redux',
        version: '3.6.0',
        description: 'Predictable state container for JavaScript apps'
      }
    ]
  }, propOverrides)

  const wrapper = shallow(<PackageList {...props} />)

  return {
    props,
    wrapper,
    refreshButton: wrapper.find('button')
  }
}

it('renders correctly', () => {
  const { wrapper } = setup()
  expect(wrapper).toMatchSnapshot()
})

it('calls its onRefresh handler', () => {
  const onRefresh = jest.fn()
  const { refreshButton } = setup({ onRefresh })

  refreshButton.simulate('click')

  expect(onRefresh.mock.calls.length).toBe(1)
})
