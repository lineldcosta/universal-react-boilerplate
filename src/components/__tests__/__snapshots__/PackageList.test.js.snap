// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="packages"
>
    <button
        onClick={[Function]}
    >
        Refresh
    </button>
    <ul>
        <li>
            react
        </li>
        <li>
            redux
        </li>
    </ul>
</div>,
  "nodes": Array [
    <div
      className="packages"
>
      <button
            onClick={[Function]}
      >
            Refresh
      </button>
      <ul>
            <li>
                  react
            </li>
            <li>
                  redux
            </li>
      </ul>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PackageList
        onRefresh={[Function]}
        packages={
                Array [
                        Object {
                          "description": "React is a JavaScript library for building user interfaces.",
                          "name": "react",
                          "version": "15.4.2",
                        },
                        Object {
                          "description": "Predictable state container for JavaScript apps",
                          "name": "redux",
                          "version": "3.6.0",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onRefresh": [Function],
          "packages": Array [
            Object {
              "description": "React is a JavaScript library for building user interfaces.",
              "name": "react",
              "version": "15.4.2",
            },
            Object {
              "description": "Predictable state container for JavaScript apps",
              "name": "redux",
              "version": "3.6.0",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="packages"
>
          <button
                    onClick={[Function]}
          >
                    Refresh
          </button>
          <ul>
                    <li>
                              react
                    </li>
                    <li>
                              redux
                    </li>
          </ul>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="packages"
>
          <button
                    onClick={[Function]}
          >
                    Refresh
          </button>
          <ul>
                    <li>
                              react
                    </li>
                    <li>
                              redux
                    </li>
          </ul>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PackageList
    onRefresh={[Function]}
    packages={
        Array [
            Object {
              "description": "React is a JavaScript library for building user interfaces.",
              "name": "react",
              "version": "15.4.2",
            },
            Object {
              "description": "Predictable state container for JavaScript apps",
              "name": "redux",
              "version": "3.6.0",
            },
          ]
    }
/>,
}
`;
